src/MongoDB.Driver/Linq/Linq3Implementation/Reflection/QueryableMethod.cs

```diff
@@ -202,7 +202,7 @@ static QueryableMethod()
             __singleOrDefault = ReflectionInfo.Method((IQueryable<object> source) => source.SingleOrDefault());
             __singleOrDefaultWithPredicate = ReflectionInfo.Method((IQueryable<object> source, Expression<Func<object, bool>> predicate) => source.SingleOrDefault(predicate));
             __singleWithPredicate = ReflectionInfo.Method((IQueryable<object> source, Expression<Func<object, bool>> predicate) => source.Single(predicate));
-            __skip = ReflectionInfo.Method((IQueryable<object> source, int count) => source.Skip(count));
+            __skip = ReflectionInfo.Method((IQueryable<object> source, int count) => Queryable.Skip(source, count));
```

Comment:
Weird method overload resolution craziness.

Even though `Queryable.Skip(IQueryable<T> source, int count` is a closer match the compiler chose `MongoQueryable.Skip(IQueryable<T> source, long count)` because it's in a closer namespace!!!